src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -4194,17 +4194,26 @@ public void visitBindingPattern(JCBindingPattern tree) {
 
     @Override
     public void visitRecordPattern(JCRecordPattern tree) {
-        Type type = attribType(tree.deconstructor, env);
-        if (type.isRaw() && type.tsym.getTypeParameters().nonEmpty()) {
-            Type inferred = infer.instantiatePatternType(resultInfo.pt, type.tsym);
-            if (inferred == null) {
-                log.error(tree.pos(), Errors.PatternTypeCannotInfer);
-            } else {
-                type = inferred;
+        Type site;
+
+        if (tree.deconstructor == null) {
```

Comment:
my question here would be, this situation can only happen if there are `var`(s) in the record pattern?

Reply:
I think so, yes:

https://github.com/openjdk/jdk/blob/master/src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavacParser.java#L846-L869

Initially I thought I could capture this situation before line 816, but then I wouldn't be able to flow an erroneous type from parsing (`types.createErrorType(tree.record.type);`).

Reply:
ok sounds good